scvi/model/_scvi.py

```diff
@@ -184,8 +206,62 @@ def setup_anndata(
                 REGISTRY_KEYS.CONT_COVS_KEY, continuous_covariate_keys
             ),
         ]
+        # register new fields for latent mode if needed
+        latent_mode = _get_latent_adata_type(adata)
+        if latent_mode is not None:
+            anndata_fields += cls._get_latent_fields(latent_mode)
         adata_manager = AnnDataManager(
             fields=anndata_fields, setup_method_args=setup_method_args
         )
         adata_manager.register_fields(adata, **kwargs)
         cls.register_manager(adata_manager)
+
+    def _get_latent_adata_from_adata(
+        self,
+        mode: LatentDataType,
+        use_latent_qzm_key: str = "X_latent_qzm",
+        use_latent_qzv_key: str = "X_latent_qzv",
+    ):
+        if mode == "dist":
+            self.adata.obsm[_SCVI_LATENT_QZM] = self.adata.obsm[use_latent_qzm_key]
+            self.adata.obsm[_SCVI_LATENT_QZV] = self.adata.obsm[use_latent_qzv_key]
+        else:
+            raise ValueError(f"Unknown latent mode: {mode}")
+        self.adata.uns[_ADATA_LATENT_UNS_KEY] = mode
+        del self.adata.raw
+        all_zeros = csr_matrix(self.adata.X.shape)
+        self.adata.X = all_zeros.copy()
+        self.adata.layers = {layer: all_zeros.copy() for layer in self.adata.layers}
+
+    @staticmethod
+    def _get_latent_fields(mode: LatentDataType):
+        if mode == "dist":
+            latent_fields = [
+                ObsmField(
+                    REGISTRY_KEYS.LATENT_QZM_KEY,
+                    _SCVI_LATENT_QZM,
+                ),
+                ObsmField(
+                    REGISTRY_KEYS.LATENT_QZV_KEY,
+                    _SCVI_LATENT_QZV,
+                ),
+            ]
+        else:
+            raise ValueError(f"Unknown latent mode: {mode}")
+        latent_fields.append(
+            StringUnsField(
+                REGISTRY_KEYS.LATENT_MODE_KEY,
+                _ADATA_LATENT_UNS_KEY,
+            ),
+        )
+        return latent_fields
+
+    def to_latent_mode(
+        self,
+        mode: LatentDataType,
+        use_latent_qzm_key: str = "X_latent_qzm",
+        use_latent_qzv_key: str = "X_latent_qzv",
+    ):
```

Comment:
can we document this? It should mention that the model.adata will be modified inplace and user should make a copy if they want to preserve it.